# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.cpp
# Commit: 7427a6e013ec
# Full Hash: 7427a6e013ec4b2101832257b904e99f9b722568
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-06-18 21:42:24
# Description:
#   Bug 1902333 - Fix crash when serializing empty shadow tree. r=smaug,dom-core,peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213628
# ==============================================================================

diff -r 65a675f06555 -r 7427a6e013ec dom/base/nsContentUtils.cpp
--- a/dom/base/nsContentUtils.cpp	Tue Jun 18 14:44:40 2024 +0000
+++ b/dom/base/nsContentUtils.cpp	Tue Jun 18 14:55:20 2024 +0000
@@ -9705,6 +9705,11 @@
   }
 
   aBuilder.Append(u">");
+
+  if (!shadow->HasChildren()) {
+    aBuilder.Append(u"</template>");
+    return false;
+  }
   return true;
 }
 